### ISSUE TITLE
Adding or removing users can cause a 503 error 

### ISSUE THREAD

**Issue by RaywanS** [gristlabs/grist-core#918]
Hi team,

I have noticed an issue while adding/removing users (both at site level and folder level).

Basically, I just add (or remove) one or multiple users and click on "Confirm". As it take forevers (average of 10 seconds), my natural behavior is to click again few times on the Confirm button, which often leads to longer waiting time and then a 503 error, as seen on the screenshot.

![bug](https://github.com/gristlabs/grist-core/assets/131388510/60fe8882-053b-43f1-999e-1e97f14c7c7b)

We were able to reproduce it with @fflorent

**Comment by RaywanS** [comment]
PS : this screenshot doesn't show the 503 error but most of the time, it displays it in the error box

**Comment by vviers** [comment]
Idea : deactivate the "Confirm" button when a request is already going on

**Comment by hexaltation** [comment]
May be linked with #1005

**Comment by manuhabitela** [comment]
Yeah, without giving it special thoughts, just setting `disabled` on a button after clicking it can cause issues with keyboard users/screen readers. For example, focus can go _poof_ and go back to the very top of the `body` tag, because focus was on something that doesn't exist anymore in the "tabbable" element list.

We could _virtually_ disable the button, but not having it actually `disabled` in the DOM, but making sure clicking doesn't trigger any action. We can take some inspiration from react-aria that has "pending" buttons for such cases: https://react-spectrum.adobe.com/react-aria/Button.html#pending
